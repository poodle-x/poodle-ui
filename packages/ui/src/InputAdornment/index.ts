import { InputAdornment, InputAdornmentProps } from "./InputAdornment";
import { StandardThemeConfig } from "../theme";

export * from "./InputAdornment";

export default InputAdornment;

export type InputAdornmentStyleKeys = "root";

export interface InputAdornmentThemeConfig
	extends StandardThemeConfig<InputAdornmentStyleKeys, InputAdornmentProps> {}
