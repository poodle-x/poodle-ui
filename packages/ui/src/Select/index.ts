import { Select, SelectProps } from "./Select";
import { StandardThemeConfig } from "../theme";

export * from "./Select";

export default Select;

export type SelectStyleKeys = "root" | "native" | "icon" | "divider";

export interface SelectThemeConfig
	extends StandardThemeConfig<SelectStyleKeys, SelectProps> {}
