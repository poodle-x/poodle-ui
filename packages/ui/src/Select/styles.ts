import { createCSSSystemStandard } from "../theme";
import { CSSObjectSystem, CSSSystem } from "../styled";
import { SelectProps } from "./Select";
import { SelectStyleKeys } from "./index";

export function Root(props: SelectProps): CSSSystem {
	const { theme, error } = props;

	let errorStyles: CSSObjectSystem = {};

	if (error) {
		errorStyles = {
			borderColor: "negative.base",

			"&:focus-within": {
				borderColor: "negative.base",
			},
			"&:hover": {
				borderColor: "negative.base",
			},
		};
	}

	return createCSSSystemStandard<SelectStyleKeys>({
		key: "root",
		config: theme?.Select,
		props,
		base: [
			{
				position: "relative",
				display: "flex",
				alignItems: "center",
				width: "100%",
				boxSizing: "border-box",
				borderRadius: "4px",
				border: "1px solid",
				borderColor: "border.base",
				backgroundColor: "#fff",
				transition: "125ms border-color ease-in",

				"&:focus-within": {
					borderColor: "primary.base",
				},
				"&:hover": {
					borderColor: "primary.base",
				},
			},
			errorStyles,
		],
	});
}

export function Native(props: SelectProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<SelectStyleKeys>({
		key: "native",
		config: theme?.Select,
		props,
		base: [
			{
				paddingX: "scale-2",
				paddingY: "scale-2.5",
				width: "100%",
				border: "none",
				outline: "none",
				boxSizing: "content-box",
				background: "transparent",
				textStyle: "input",
				appearance: "none",

				"&:disabled": {
					cursor: "not-allowed",
					color: "disabledText",
				},
			},
		],
	});
}

export function Icon(props: SelectProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<SelectStyleKeys>({
		key: "icon",
		config: theme?.Select,
		props,
		base: [
			{
				display: "flex",
				paddingRight: "scale-2",
				height: "100%",
				alignItems: "center",
				fontSize: "1.5rem",
				pointerEvents: "none",
			},
		],
	});
}

export function Divider(props: SelectProps): CSSSystem {
	const { theme } = props;

	return createCSSSystemStandard<SelectStyleKeys>({
		key: "divider",
		config: theme?.Select,
		props,
		base: [
			{
				display: "flex",
				width: "1px",
				height: "1rem",
				backgroundColor: "border.base",
				marginRight: "scale-2",
			},
		],
	});
}
