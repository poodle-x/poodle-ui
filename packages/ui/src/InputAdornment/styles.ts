import { createCSSSystemStandard } from "../theme";
import { CSSObjectSystem, CSSSystem } from "../styled";
import { InputAdornmentProps } from "./InputAdornment";
import { InputAdornmentStyleKeys } from "./";

export function Root(props: InputAdornmentProps): CSSSystem {
	const { theme, adornmentPosition, isIcon } = props;

	let positionStyles: CSSObjectSystem = {};
	let iconStyles: CSSObjectSystem = {};

	if (adornmentPosition === "start") {
		positionStyles = {
			paddingLeft: "scale-2",
		};
	} else {
		positionStyles = {
			paddingRight: "scale-2",
		};
	}

	if (isIcon) {
		iconStyles = {
			fontSize: "1.5rem",
		};
	}

	return createCSSSystemStandard<InputAdornmentStyleKeys>({
		key: "root",
		config: theme?.InputAdornment,
		props,
		base: [
			{
				display: "flex",
				color: "text",
				alignItems: "center",
				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
				// @ts-ignore
				textStyle: "input",
				...(positionStyles as CSSObjectSystem),
				...iconStyles,
			},
		],
	});
}
