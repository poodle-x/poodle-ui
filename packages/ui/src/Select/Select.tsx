import React from "react";
import Box, { BoxProps } from "../Box";
import { StandardComponentProps, ThemeConfig } from "../theme";
import useDefaultProps from "../utils/useDefaultProps";
import { useClassNames } from "../styled";
import * as styles from "./styles";
import ArrowDown from "../icons/ArrowDown";

export interface LocalSelectProps {
	/**
	 * Change the component style to error style when set to true.
	 */
	error?: boolean;
	/**
	 * Add prefix to the input.
	 */
	startAdornment?: React.ReactNode;
	/**
	 * Add suffix to the input.
	 */
	endAdornment?: React.ReactNode;
	/**
	 * Custom select props
	 */
	selectProps?: BoxProps;
	/**
	 * Custom icon props
	 */
	iconProps?: BoxProps;
	/**
	 * Custom icon props
	 */
	dividerProps?: BoxProps;
}

export interface SelectProps
	extends BoxProps,
		StandardComponentProps,
		LocalSelectProps {}

function getDefaultProps(theme?: ThemeConfig) {
	return theme?.Select?.defaultProps;
}

export const Select: React.ForwardRefExoticComponent<
	React.PropsWithoutRef<SelectProps> & React.RefAttributes<HTMLElement>
> = React.forwardRef<HTMLElement, SelectProps>((_props, ref) => {
	const props = useDefaultProps<SelectProps>(_props, {
		themeDefaultProps: getDefaultProps,
	});

	const {
		children,
		className,
		error,
		selectProps,
		iconProps,
		dividerProps,
		startAdornment,
		endAdornment,
		...otherProps
	} = props;

	const classes = useClassNames({
		props,
		lists: {
			root: {
				classNames: ["poodle-select", styles.Root, className],
			},
			native: {
				classNames: [
					"poodle-select__native",
					styles.Native,
					selectProps?.className,
				],
			},
			icon: {
				classNames: ["poodle-select__icon", styles.Icon, iconProps?.className],
			},
			divider: {
				classNames: [
					"poodle-select__divider",
					styles.Divider,
					dividerProps?.className,
				],
			},
		},
	});

	const ArrowDownIcon = props.theme?.baseIcons?.ArrowDown
		? props.theme?.baseIcons?.ArrowDown
		: ArrowDown;

	return (
		<Box {...otherProps} className={classes.root} ref={ref}>
			{startAdornment && startAdornment}
			<Box
				as="select"
				aria-invalid={error}
				{...selectProps}
				className={classes.native}
			>
				{children}
			</Box>
			<Box {...iconProps} className={classes.icon}>
				<ArrowDownIcon />
			</Box>
			{endAdornment && <Box {...dividerProps} className={classes.divider} />}
			{endAdornment}
		</Box>
	);
});

Select.displayName = "PoodleSelect";
