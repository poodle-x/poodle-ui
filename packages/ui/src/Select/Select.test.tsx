import React from "react";
import { render } from "../utils/test";
import Select from "./";
import { ThemeConfig } from "../theme";
import { InputAdornment } from "../InputAdornment";

describe("<Select />", () => {
	it("should render Select correctly", () => {
		const { container } = render(
			<Select>
				<option value="tiger">Tiger</option>
				<option value="lion">Lion</option>
				<option value="dog">Dog</option>
			</Select>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render Select with error correctly", () => {
		const { container } = render(
			<Select error={true}>
				<option value="tiger">Tiger</option>
				<option value="lion">Lion</option>
				<option value="dog">Dog</option>
			</Select>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("should render Select with adornment correctly", () => {
		const { container } = render(
			<Select
				error={true}
				startAdornment={<InputAdornment>$</InputAdornment>}
				endAdornment={
					<InputAdornment
						adornmentPosition="end"
						isIcon={true}
						color="negative.base"
					>
						end
					</InputAdornment>
				}
			>
				<option value="tiger">Tiger</option>
				<option value="lion">Lion</option>
				<option value="dog">Dog</option>
			</Select>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	it("ref should work", () => {
		const ref = React.createRef<HTMLDivElement>();

		render(
			<Select ref={ref}>
				<option value="tiger">Tiger</option>
				<option value="lion">Lion</option>
				<option value="dog">Dog</option>
			</Select>
		);
		expect(ref.current).toMatchSnapshot();
	});

	it("should render Box props correctly", () => {
		const { container } = render(
			<Select
				id="id"
				mt="20px"
				selectProps={{
					"aria-label": "name",
				}}
			>
				<option value="tiger">Tiger</option>
				<option value="lion">Lion</option>
				<option value="dog">Dog</option>
			</Select>
		);
		expect(container.firstChild).toMatchSnapshot();
	});

	describe("theming", () => {
		const themeTest: ThemeConfig = {
			mode: "mode",
			Select: {
				styles: {
					root: {
						mt: 10,
						letterSpacing: 250,
					},
					icon: {
						mt: 11,
						letterSpacing: 251,
					},
					divider: {
						mt: 12,
						letterSpacing: 252,
					},
					native: {
						mt: 13,
						letterSpacing: 251,
					},
				},
				variants: {
					custom: {
						root: {
							top: 0,
							textIndent: 1,
						},
						icon: {
							top: 1,
							textIndent: 2,
						},
						native: {
							top: 2,
							textIndent: 3,
						},
						divider: {
							top: 3,
							textIndent: 4,
						},
					},
				},
			},
		};

		it("should extend styles in global", () => {
			const { container } = render(
				<Select variant="custom">
					<option value="tiger">Tiger</option>
					<option value="lion">Lion</option>
					<option value="dog">Dog</option>
				</Select>,
				{
					theme: themeTest,
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should extend styles in local", () => {
			const { container } = render(
				<Select variant="custom" themeExtend={themeTest}>
					<option value="tiger">Tiger</option>
					<option value="lion">Lion</option>
					<option value="dog">Dog</option>
				</Select>
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should render with default props correctly", () => {
			const { container } = render(
				<Select>
					<option value="tiger">Tiger</option>
					<option value="lion">Lion</option>
					<option value="dog">Dog</option>
				</Select>,
				{
					theme: {
						Select: {
							defaultProps: {
								id: "id",
								className: "hello",
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should disable classes", () => {
			const { container } = render(
				<Select>
					<option value="tiger">Tiger</option>
					<option value="lion">Lion</option>
					<option value="dog">Dog</option>
				</Select>,
				{
					theme: {
						Select: {
							overrideClasses: {
								root: true,
								divider: true,
								native: true,
								icon: true,
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should override classes", () => {
			const { container } = render(
				<Select>
					<option value="tiger">Tiger</option>
					<option value="lion">Lion</option>
					<option value="dog">Dog</option>
				</Select>,
				{
					theme: {
						Select: {
							overrideClasses: {
								root: "morning",
								divider: "divider",
								icon: "icon",
								native: "native",
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});

		it("should override styles", () => {
			const { container } = render(
				<Select>
					<option value="tiger">Tiger</option>
					<option value="lion">Lion</option>
					<option value="dog">Dog</option>
				</Select>,
				{
					theme: {
						Select: {
							overrides: {
								root: {
									mt: 51,
								},
								divider: {
									mt: 52,
								},
								icon: {
									mt: 53,
								},
								native: {
									mt: 54,
								},
							},
						},
					},
				}
			);
			expect(container.firstChild).toMatchSnapshot();
		});
	});
});
