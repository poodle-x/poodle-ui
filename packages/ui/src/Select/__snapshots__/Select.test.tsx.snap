// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> ref should work 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.c4:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
>
  <select
    class="poodle-box c0 poodle-select__native c4 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> should render Box props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c2 {
  margin-top: 20px;
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.c4:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
  id="id"
>
  <select
    aria-label="name"
    class="poodle-box c0 poodle-select__native c4 c5"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c5"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> should render Select correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.c4:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
>
  <select
    class="poodle-box c0 poodle-select__native c4 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> should render Select with adornment correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  border-color: #AF1308;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c1:focus-within {
  border-color: #AF1308;
}

.c1:hover {
  border-color: #AF1308;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #282828;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  padding-left: 8px;
}

.c7 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.c7:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 1px;
  height: 1rem;
  background-color: #CCCCCC;
  margin-right: 8px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #282828;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  padding-right: 8px;
  font-size: 1.5rem;
}

.c17 {
  color: #AF1308;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
>
  <div
    class="poodle-box c0 poodle-input-adornment c4 c2"
  >
    $
  </div>
  <select
    aria-invalid="true"
    class="poodle-box c0 poodle-select__native c7 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c10 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
  <div
    class="poodle-box c0 poodle-select__divider c13 c2"
  />
  <div
    class="poodle-box c0 poodle-input-adornment c16 c17"
  >
    end
  </div>
</div>
`;

exports[`<Select /> should render Select with error correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  border-color: #AF1308;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c1:focus-within {
  border-color: #AF1308;
}

.c1:hover {
  border-color: #AF1308;
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.c4:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
>
  <select
    aria-invalid="true"
    class="poodle-box c0 poodle-select__native c4 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> theming should disable classes 1`] = `
.c0 {
  box-sizing: border-box;
}

<div
  class="poodle-box c0 poodle-select c1"
>
  <select
    class="poodle-box c0 poodle-select__native c1"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> theming should extend styles in global 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin-top: 13px;
  letter-spacing: 251px;
  top: 2px;
  text-indent: 3px;
}

.c4:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
  margin-top: 11px;
  letter-spacing: 251px;
  top: 1px;
  text-indent: 2px;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
>
  <select
    class="poodle-box c0 poodle-select__native c4 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> theming should extend styles in local 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
  margin-top: 10px;
  letter-spacing: 250px;
  top: 0;
  text-indent: 1px;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  margin-top: 13px;
  letter-spacing: 251px;
  top: 2px;
  text-indent: 3px;
}

.c4:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
  margin-top: 11px;
  letter-spacing: 251px;
  top: 1px;
  text-indent: 2px;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
>
  <select
    class="poodle-box c0 poodle-select__native c4 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> theming should override classes 1`] = `
.c0 {
  box-sizing: border-box;
}

<div
  class="poodle-box c0 poodle-select morning c1"
>
  <select
    class="poodle-box c0 poodle-select__native native c1"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon icon c1"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> theming should override styles 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  margin-top: 51px;
}

.c4 {
  margin-top: 54px;
}

.c7 {
  margin-top: 53px;
}

<div
  class="poodle-box c0 poodle-select c1 c2"
>
  <select
    class="poodle-box c0 poodle-select__native c4 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;

exports[`<Select /> theming should render with default props correctly 1`] = `
.c0 {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid;
  border-color: #CCCCCC;
  background-color: #fff;
  -webkit-transition: 125ms border-color ease-in;
  transition: 125ms border-color ease-in;
}

.c1:focus-within {
  border-color: #00478A;
}

.c1:hover {
  border-color: #00478A;
}

.c4 {
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 100%;
  border: none;
  outline: none;
  box-sizing: content-box;
  background: transparent;
  font-size: 1rem;
  font-weight: 400;
  line-height: 125%;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
}

.c4:disabled {
  cursor: not-allowed;
  color: #A6A6A6;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 8px;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.5rem;
  pointer-events: none;
}

<div
  class="poodle-box c0 poodle-select c1 hello c2"
  id="id"
>
  <select
    class="poodle-box c0 poodle-select__native c4 c2"
  >
    <option
      value="tiger"
    >
      Tiger
    </option>
    <option
      value="lion"
    >
      Lion
    </option>
    <option
      value="dog"
    >
      Dog
    </option>
  </select>
  <div
    class="poodle-box c0 poodle-select__icon c7 c2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="m12 16 6.928-6H5.072L12 16z"
      />
    </svg>
  </div>
</div>
`;
